src/main/groovy/pluggable/scm/bitbucket/BitbucketRequestUtil.groovy

```diff
@@ -0,0 +1,106 @@
+package pluggable.scm.bitbucket;
+
+import pluggable.scm.helpers.*;
+import java.net.URL;
+import java.io.IOException;
+import java.io.DataOutputStream;
+import java.nio.charset.Charset;
+import groovy.json.JsonSlurper;
+import groovy.json.JsonOutput;
+
+
+public class BitbucketRequestUtil {
+
+  public static void isProjectAvailable(URL bitbucketUrl, String username, String password, String projectKey){
+
+    URL url = new URL(bitbucketUrl, "/rest/api/latest/projects/" + projectKey);
+    def auth = "${username}:${password}".bytes.encodeBase64().toString();
+
+    HttpURLConnection http = (HttpURLConnection) url.openConnection();
+    http.setRequestMethod("GET");
+    http.setRequestProperty ("Authorization", "Basic ${auth}");
+
+    switch (http.getResponseCode()) {
+      case 200:
+        Logger.info("Project ${projectKey} found all good!");
+        break;
+      case 401:
+        Logger.log(LogLevel.ERROR, "Credentials are invalid.");
+        break;
+      case {it > 401}:
+        throw new IOException("Project doesn't exist or host not available!");
+        break;
+    }
+  }
+
+  public static String[] getProjectRepositorys(URL bitbucketUrl, String username, String password, String projectKey){
+
+    JsonSlurper jsonSlurper = new JsonSlurper();
+    URL url = new URL(bitbucketUrl, "/rest/api/latest/projects/${projectKey}/repos");
+    def auth = "${username}:${password}".bytes.encodeBase64().toString();
+    List<String> repositoryList = [];
+
+    HttpURLConnection http = (HttpURLConnection) url.openConnection();
+    http.setRequestMethod("GET");
+    http.setRequestProperty ("Authorization", "Basic ${auth}");
+    http.setRequestProperty("Content-Type", "application/json");
+
+    switch (http.getResponseCode()) {
+      case 200:
+        def json = jsonSlurper.parse(http.getInputStream())
+        for(int i = 0; i < json.size; i++){
+          repositoryList.add(json.values[i].name)
```

Comment:
Good shout! I've defaulted the limit to 100. We will add this as an issue to support pagination.